03week/ticTacToe.js

```diff
@@ -23,27 +30,64 @@ function printBoard() {
   console.log('2 ' + board[2].join(' | '));
 }
 
-function horizontalWin() {
-  // Your code here
+const isPlayerTurn = (val) => {
+  return val === playerTurn;
 }
 
-function verticalWin() {
-  // Your code here
+const horizontalWin = () => {
+  if(board[0].every(isPlayerTurn) || board[1].every(isPlayerTurn) || board[2].every(isPlayerTurn)){
+    console.log(`${playerTurn} wins!`)
+    return true;
+  }
 }
 
-function diagonalWin() {
-  // Your code here
-}
 
-function checkForWin() {
-  // Your code here
+const verticalWin = () => {
+  if(board[0][0] && board[1][0] && board[2][0] == playerTurn  ||board[0][1] && board[1][1] && board[2][1] == playerTurn || board[0][2] && board[1][2] && board[2][2] == playerTurn){
+    console.log(`${playerTurn} wins!`)
+    return true;
+  }
 }
 
-function ticTacToe(row, column) {
-  // Your code here
+const diagonalWin = () => {
+  if(board[0][0] && board[1][1] && board[2][2] == playerTurn || board[0][2] && board[1][1] && board[2][0] == playerTurn){
+    console.log(`${playerTurn} wins!`)
+    return true;
+  }
 }
 
-function getPrompt() {
+const checkForWin = () => {
+  return horizontalWin() || verticalWin() || diagonalWin()  
+};
+
+const isValidMove = (row, column) => {
+  return board[row][column] === ' '
+};
+
+const switchPlayer = () => {
+  if(playerTurn == 'X'){
+    playerTurn = 'O'
+  } else {
+    playerTurn = 'X'
+  }
+};
+
+// const resetBoard = () => {
+//   board[0].splice(0, 3);
+//   board[1].splice(0, 3);
+//   board[2].splice(0, 3);
+// }
+
+const ticTacToe = (row, column) => {
+  if(isValidMove(row, column)){
```

Comment:
in this function you have something that are not in order and functionally is giving you errors if you try to put the an O or X in a spot that is already exists.